pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -444,14 +444,14 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 			// filter the changes in workflow status
 			// let workflow handle its reconcile
 			UpdateFunc: func(e ctrlEvent.UpdateEvent) bool {
-				new, ok := e.ObjectNew.DeepCopyObject().(*v1beta1.Application)
-				if !ok {
-					return true
-				}
-				old, ok := e.ObjectOld.DeepCopyObject().(*v1beta1.Application)
-				if !ok {
-					return true
+				new, isNewApp := e.ObjectNew.DeepCopyObject().(*v1beta1.Application)
+				old, isOldApp := e.ObjectOld.DeepCopyObject().(*v1beta1.Application)
+				if !isNewApp || !isOldApp {
+					return filterManagedFieldChangesUpdate(e)
 				}
+				// filter managedFields changes
+				new.ManagedFields = old.ManagedFields
```

Comment:
managedFields could change when generation not changed?

Reply:
Yes. k8s 1.18.5 will do that.